upgrade.yml

```diff
@@ -18,6 +18,16 @@
         openstack_source:
           git_update: yes
 
+    - name: set ssh_service fact
+      set_fact:
+        ssh_service: "{{ (ursula_os == 'ubuntu') | ternary('ssh','sshd') }}"
+
+    - name: set ursula_os if undefined
+      set_fact:
+        ursula_os: "{{ ansible_distribution | regex_replace('RedHat|CentOS', 'rhel')|lower }}"
+      when: ( ursula_os is undefined ) and
+            ( ansible_distribution in ['CentOS', 'RedHat'] or ansible_distribution == 'Ubuntu')
+
```

Comment:
seems like the second stanza 'set ursula_os if undefined' should go before 'set ssh_service fact' since it relies on ursula_os ?